File: packages/android-im-ui/src/test/kotlin/com/flare/im/ui/ConversationKindTest.kt
package com.flare.im.ui

import java.io.File
import kotlin.test.Test
import kotlin.test.assertEquals

/** FR-056: one conversation-kind vocabulary for every kit and every component. */
class ConversationKindTest {
    private fun vocabularyFile(): File {
        var dir: File? = File(".").absoluteFile
        while (dir != null) {
            val candidate = File(dir, "spec/conversation-kind.json")
            if (candidate.isFile) return candidate
            dir = dir.parentFile
        }
        error("spec/conversation-kind.json not found")
    }

    @Test fun theKitEnumIsTheSharedVocabulary() {
        val kinds = (FlareJson.parse(vocabularyFile().readText()) as Map<*, *>)["kinds"]
        assertEquals(kinds, FlareConversationKind.entries.map { it.name.replaceFirstChar { c -> c.lowercase() } })
    }
}

File: packages/flutter-im-ui/test/conversation_kind_test.dart
import 'dart:convert';
import 'dart:io';

import 'package:flare_im_ui/flare_im_ui.dart';
import 'package:flutter_test/flutter_test.dart';

// FR-056: one conversation-kind vocabulary for every kit and every component.

void main() {
  test('the kit enum is the shared vocabulary', () {
    final vocabulary =
        jsonDecode(
              File(
                '${Directory.current.path}/../../spec/conversation-kind.json',
              ).readAsStringSync(),
            )
            as Map<String, dynamic>;
    expect(
      FlareConversationKind.values.map((kind) => kind.name).toList(),
      (vocabulary['kinds'] as List).cast<String>(),
    );
  });
}
